tests/test_modeling_gpt2.py

```diff
@@ -37,6 +37,9 @@
 class GPT2ModelTest(ModelTesterMixin, unittest.TestCase):
 
     all_model_classes = (GPT2Model, GPT2LMHeadModel, GPT2DoubleHeadsModel) if is_torch_available() else ()
+    all_generative_model_classes = (
+        (GPT2LMHeadModel,) if is_torch_available() else ()
+    )  # TODO (PVP): Add Double HeadsModel when generate() function is changed accordingly
```

Comment:
The `GPT2DoubleHeadsModel` has the hidden states cache at a different position in the output (`=output[2]`) than `GPT2LMHeadModel` `(output[1])` which would in the current version of the language generation throw an error. As discussed with @thomwolf all the models should return [named tuples](https://docs.python.org/3/library/collections.html#collections.namedtuple) in the near future which would greatly simplify getting the "correct" outputs for the lm model generation function. I will treat this in a following PR. @sshleifer, @LysandreJik

Reply:
Sounds great. Named tuples (and named tensors) are very exciting.